app/src/main/java/org/dicekeys/app/fragments/AssembleFragment.kt

```diff
@@ -1,19 +1,220 @@
 
+import android.animation.ObjectAnimator
+import android.animation.ValueAnimator
 import android.os.Bundle
+import android.view.LayoutInflater
 import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.ImageView
+import androidx.annotation.LayoutRes
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.FragmentStatePagerAdapter
+import androidx.fragment.app.viewModels
+import androidx.navigation.NavOptions
+import androidx.navigation.NavOptionsBuilder
+import androidx.navigation.fragment.findNavController
+import androidx.viewpager.widget.ViewPager
 import dagger.hilt.android.AndroidEntryPoint
 import org.dicekeys.app.AppFragment
+import org.dicekeys.app.NavGraphDirections
 import org.dicekeys.app.R
-import org.dicekeys.app.databinding.AssembleFragmentBinding
-import org.dicekeys.app.viewmodels.DiceKeyViewModel
+import org.dicekeys.app.databinding.*
+import org.dicekeys.app.extensions.clearNavigationResult
+import org.dicekeys.app.extensions.getNavigationResult
+import org.dicekeys.app.fragments.backup.BackupFragment
+import org.dicekeys.app.repositories.DiceKeyRepository
+import org.dicekeys.app.viewmodels.AssembleViewModel
+import org.dicekeys.app.views.DiceKeyView
+import org.dicekeys.app.views.StickerTargetSheetView
+import org.dicekeys.dicekey.DiceKey
+import org.dicekeys.dicekey.Face
+import org.dicekeys.dicekey.FaceRead
+import javax.inject.Inject
 
 @AndroidEntryPoint
-class AssembleFragment: AppFragment<AssembleFragmentBinding>(R.layout.assemble_fragment) {
+class AssembleFragment : AppFragment<AssembleFragmentBinding>(R.layout.assemble_fragment) {
 
+    @Inject
+    lateinit var diceKeyRepository: DiceKeyRepository
+
+    private lateinit var pagerAdapter: AssemblePagerAdapter
+
+
+    val viewModel: AssembleViewModel by viewModels()
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
+        binding.vm = viewModel
+
+        getNavigationResult<String>(ScanFragment.READ_DICEKEY)?.observe(viewLifecycleOwner) {
+            it?.let {
+                clearNavigationResult(ScanFragment.READ_DICEKEY)
+
+                FaceRead.diceKeyFromJsonFacesRead(it)?.let { faceRead ->
+
+                    DiceKey(faces = faceRead.faces.map {
+                        Face(letter = it.letter, digit = it.digit, orientationAsLowercaseLetterTrbl = it.orientationAsLowercaseLetterTrbl)
+                    }).also {
+                        viewModel.setDiceKey(it)
+                    }
+                }
+            }
+        }
+
+        getNavigationResult<Boolean>(BackupFragment.VALID_BACKUP)?.observe(viewLifecycleOwner) {
+            if (it) {
+                viewModel.diceKeyBackedUp.postValue(true)
+            }
+        }
+
+        pagerAdapter = AssemblePagerAdapter(childFragmentManager)
+
+        binding.viewPager.also {
```

Comment:
A little note as to what this is doing would be helpful.  Is this ensuring nothing happens on a scroll?